### PR TITLE
Add a voter for tl_undo

### PR DESCRIPTION
**Pull request by aschempp**
The voter now correctly checks permissions on the records to be restored. If a user deleted something from news but no longer has permissions to edit news, they will no longer be able to restore that record.

### CODE REVIEW

File: core-bundle/config/listener.yaml [hunk 1]
```diff
@@ -188,19 +188,25 @@ services:
...
+            - '@security.helper'
 
     contao.listener.data_container.undo.label:
         class: Contao\CoreBundle\EventListener\DataContainer\Undo\LabelListener
         arguments:
             - '@contao.framework'
             - '@twig'
 
+    contao.listener.data_container.undo_operation:
+        class: Contao\CoreBundle\EventListener\DataContainer\Undo\UndoOperationListener
```
Comment (leofeyer) [inline core-bundle/config/listener.yaml:206]:
```suggestion
        class: Contao\CoreBundle\EventListener\DataContainer\Undo\OperationListener
```

At least we didn‘t call the `Contao\CoreBundle\EventListener\DataContainer\Undo\LabelListener` class `UndoLabelListener`.

Comment (aschempp) [reply core-bundle/config/listener.yaml:206]: yeah but the `JumpToParentOperationListener` exists as well. `Undo` in `UndoOperation` represents the operation name, not the table.

File: core-bundle/src/EventListener/DataContainer/Undo/UndoOperationListener.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+            return;
+        }
+
+        foreach ($data as $table => $fields) {
+            foreach ($fields as $row) {
+                if (!$this->security->isGranted(ContaoCorePermissions::DC_PREFIX.$table, new CreateAction($table, $row))) {
+                    $operation->disable();
+
+                    return;
```
Comment (leofeyer) [inline core-bundle/src/EventListener/DataContainer/Undo/UndoOperationListener.php:47]:
```suggestion
                    continue;
```

Or did you really mean `return`?

Comment (aschempp) [reply core-bundle/src/EventListener/DataContainer/Undo/UndoOperationListener.php:47]: if the operation is disabled, there's nothing else to do.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @aschempp.

**Pull request merged**